In [1]:
import pandas as pd

In [2]:
df_country_vaccinations = pd.read_csv("country_vaccinations.csv")

In [3]:
df_country_vaccinations

country iso_code        date  total_vaccinations  people_vaccinated  \
0     Afghanistan      AFG  2021-02-22                 0.0                0.0   
1     Afghanistan      AFG  2021-02-23                 NaN                NaN   
2     Afghanistan      AFG  2021-02-24                 NaN                NaN   
3     Afghanistan      AFG  2021-02-25                 NaN                NaN   
4     Afghanistan      AFG  2021-02-26                 NaN                NaN   
...           ...      ...         ...                 ...                ...   
8446     Zimbabwe      ZWE  2021-03-23             44135.0            44135.0   
8447     Zimbabwe      ZWE  2021-03-24             49404.0            49404.0   
8448     Zimbabwe      ZWE  2021-03-25             54892.0            54892.0   
8449     Zimbabwe      ZWE  2021-03-26             61093.0            61093.0   
8450     Zimbabwe      ZWE  2021-03-27             65466.0            65466.0   

      people_fully_vaccinated  daily_vaccinations_raw  daily_vaccinations  \
0                         NaN                     NaN                 NaN   
1                         NaN                     NaN              1367.0   
2                         NaN                     NaN              1367.0   
3                         NaN                     NaN              1367.0   
4                         NaN                     NaN              1367.0   
...                       ...                     ...                 ...   
8446                      NaN                   841.0               655.0   
8447                      NaN                  5269.0              1400.0   
8448                      NaN                  5488.0              2127.0   
8449                      NaN                  6201.0              2772.0   
8450                      NaN                  4373.0              3322.0   

      total_vaccinations_per_hundred  people_vaccinated_per_hundred  \
0                               0.00                           0.00   
1                                NaN                            NaN   
2                                NaN                            NaN   
3                                NaN                            NaN   
4                                NaN                            NaN   
...                              ...                            ...   
8446                            0.30                           0.30   
8447                            0.33                           0.33   
8448                            0.37                           0.37   
8449                            0.41                           0.41   
8450                            0.44                           0.44   

      people_fully_vaccinated_per_hundred  daily_vaccinations_per_million  \
0                                     NaN                             NaN   
1                                     NaN                            35.0   
2                                     NaN                            35.0   
3                                     NaN                            35.0   
4                                     NaN                            35.0   
...                                   ...                             ...   
8446                                  NaN                            44.0   
8447                                  NaN                            94.0   
8448                                  NaN                           143.0   
8449                                  NaN                           187.0   
8450                                  NaN                           224.0   

                vaccines                source_name  \
0     Oxford/AstraZeneca  Government of Afghanistan   
1     Oxford/AstraZeneca  Government of Afghanistan   
2     Oxford/AstraZeneca  Government of Afghanistan   
3     Oxford/AstraZeneca  Government of Afghanistan   
4     Oxford/AstraZeneca  Government of Afghanistan   
... 

### I also did total number of vaccinations cause for example in China, the number of people vaccinated is zero but their total vaccination is so big.
### The data is medyo confusing :<, not sure which would be better

## Number of People Vaccinated

In [4]:
df_country_people_vaccinated = df_country_vaccinations.groupby(["country"], as_index = False)["people_vaccinated"].max()

In [5]:
df_country_people_vaccinated

country  people_vaccinated
0    Afghanistan            54000.0
1        Albania             6073.0
2        Algeria                NaN
3        Andorra             7098.0
4         Angola            87022.0
..           ...                ...
149      Uruguay           509672.0
150    Venezuela            12194.0
151      Vietnam            44000.0
152        Wales          1365355.0
153     Zimbabwe            65466.0

[154 rows x 2 columns]

In [6]:
country_list = list(df_country_people_vaccinated["country"])

In [7]:
df_country_people_vaccinated.to_csv("country_people_vaccinated.csv")

In [8]:
df_temp_people_vaccinated = df_country_vaccinations.groupby(["country", "vaccines"], as_index = False)["people_vaccinated"].max()

In [9]:
df_temp_people_vaccinated

country                             vaccines  people_vaccinated
0    Afghanistan                   Oxford/AstraZeneca            54000.0
1        Albania                      Pfizer/BioNTech             6073.0
2        Algeria                            Sputnik V                NaN
3        Andorra                      Pfizer/BioNTech             7098.0
4         Angola                   Oxford/AstraZeneca            87022.0
..           ...                                  ...                ...
149      Uruguay             Pfizer/BioNTech, Sinovac           509672.0
150    Venezuela                            Sputnik V            12194.0
151      Vietnam                   Oxford/AstraZeneca            44000.0
152        Wales  Oxford/AstraZeneca, Pfizer/BioNTech          1365355.0
153     Zimbabwe                    Sinopharm/Beijing            65466.0

[154 rows x 3 columns]

In [10]:
df_vaccine_people_vaccinated = df_temp_people_vaccinated.groupby(["vaccines"], as_index = False)["people_vaccinated"].sum()

In [11]:
df_vaccine_people_vaccinated

vaccines  people_vaccinated
0                         Covaxin, Oxford/AstraZeneca         51441436.0
1                             EpiVacCorona, Sputnik V          6300000.0
2                                     Johnson&Johnson           231605.0
3           Johnson&Johnson, Moderna, Pfizer/BioNTech         91707729.0
4                                             Moderna            91918.0
5                         Moderna, Oxford/AstraZeneca            43073.0
6        Moderna, Oxford/AstraZeneca, Pfizer/BioNTech         47111432.0
7   Moderna, Oxford/AstraZeneca, Pfizer/BioNTech, ...          1903498.0
8                            Moderna, Pfizer/BioNTech          7930532.0
9                                  Oxford/AstraZeneca          8056185.0
10                Oxford/AstraZeneca, Pfizer/BioNTech         61809097.0
11  Oxford/AstraZeneca, Pfizer/BioNTech, Sinopharm...          3480415.0
12  Oxford/AstraZeneca, Pfizer/BioNTech, Sinopharm...          1844536.0
13       Oxford/AstraZeneca, Pfizer/BioNTech, Sinovac            53886.0
14     Oxford/AstraZeneca, Pfizer/BioNTech, Sputnik V          5901505.0
15              Oxford/AstraZeneca, Sinopharm/Beijing          4365315.0
16   Oxford/AstraZeneca, Sinopharm/Beijing, Sputnik V          2981814.0
17                        Oxford/AstraZeneca, Sinovac         14891727.0
18                                    Pfizer/BioNTech          3090925.0
19                 Pfizer/BioNTech, Sinopharm/Beijing           296105.0
20                           Pfizer/BioNTech, Sinovac         17442038.0
21                         Pfizer/BioNTech, Sputnik V            38853.0
22                                  Sinopharm/Beijing          1106923.0
23        Sinopharm/Beijing, Sinopharm/Wuhan, Sinovac                0.0
24                       Sinopharm/Beijing, Sputnik V            55582.0
25                                            Sinovac          7701083.0
26                                          Sputnik V           344530.0

In [12]:
vaccine_people_vaccinated = dict()
for vaccine in df_vaccine_people_vaccinated["vaccines"]: 
    if "," not in vaccine: 
        vaccine_people_vaccinated[vaccine] = df_vaccine_people_vaccinated.loc[df_vaccine_people_vaccinated["vaccines"] == vaccine, "people_vaccinated"].iloc[0]
    else: 
        vaccine_list = vaccine.split(",")
        for sub_vaccine in vaccine_list: 
            sub_vaccine = sub_vaccine.strip()
            if sub_vaccine in vaccine_people_vaccinated.keys(): 
                vaccine_people_vaccinated[sub_vaccine] = vaccine_people_vaccinated[sub_vaccine] + df_vaccine_people_vaccinated.loc[df_vaccine_people_vaccinated["vaccines"] == vaccine, "people_vaccinated"].iloc[0]
            else: 
                vaccine_people_vaccinated[sub_vaccine] = df_vaccine_people_vaccinated.loc[df_vaccine_people_vaccinated["vaccines"] == vaccine, "people_vaccinated"].iloc[0]
               
        

In [13]:
vaccine_people_vaccinated

{'Covaxin': 51441436.0,
 'Oxford/AstraZeneca': 103384480.0,
 'EpiVacCorona': 6300000.0,
 'Sputnik V': 344530.0,
 'Johnson&Johnson': 91939334.0,
 'Moderna': 57080453.0,
 'Pfizer/BioNTech': 20867921.0,
 'Sinopharm/Beijing': 1162505.0,
 'Sinopharm/Wuhan': 3480415.0,
 'Sinovac': 7701083.0}

In [14]:
vaccine_pv_col1 = list(vaccine_people_vaccinated.keys())
vaccine_pv_col2 = list(vaccine_people_vaccinated.values())

In [15]:
data_vaccine_pv = {"vaccines": vaccine_pv_col1, "people_vaccinated": vaccine_pv_col2}
df_total_vaccine_pv = pd.DataFrame.from_dict(data_vaccine_pv)

In [16]:
df_total_vaccine_pv

vaccines  people_vaccinated
0             Covaxin         51441436.0
1  Oxford/AstraZeneca        103384480.0
2        EpiVacCorona          6300000.0
3           Sputnik V           344530.0
4     Johnson&Johnson         91939334.0
5             Moderna         57080453.0
6     Pfizer/BioNTech         20867921.0
7   Sinopharm/Beijing          1162505.0
8     Sinopharm/Wuhan          3480415.0
9             Sinovac          7701083.0

In [17]:
# vaccine_people_vaccinated = dict()
# for vaccine in df_vaccine_people_vaccinated["vaccines"]: 
#     if "," not in vaccine: 
#         vaccine_people_vaccinated[vaccine] = df_vaccine_people_vaccinated.loc[df_vaccine_people_vaccinated["vaccines"] == vaccine, "people_vaccinated"].iloc[0]
# for vaccine in df_vaccine_people_vaccinated["vaccines"]: 
#     if "," in vaccine: 
#         vaccine_list = vaccine.split(",")
#         total_vaccinated = df_vaccine_people_vaccinated.loc[df_vaccine_people_vaccinated["vaccines"] == vaccine, "people_vaccinated"].iloc[0]
#         print(total_vaccinated)
#         for sub_vaccine in vaccine_list: 
#             if sub_vaccine in vaccine_people_vaccinated.keys(): 
#                 total_vaccinated = total_vaccinated - df_vaccine_people_vaccinated.loc[df_vaccine_people_vaccinated["vaccines"] == sub_vaccine, "people_vaccinated"].iloc[0]
#         print(total_vaccinated)
#         for sub_vaccine in vaccine_list: 
#             if sub_vaccine not in vaccine_people_vaccinated.keys(): 
#                 vaccine_people_vaccinated[sub_vaccine] = total_vaccinated

#         for sub_vaccine in vaccine_list: 
#             if sub_vaccine in vaccine_people_vaccinated.keys(): 
#                 total_vaccinated = total_vaccinated - df_vaccine_people_vaccinated.loc[df_vaccine_people_vaccinated["vaccines"] == sub_vaccine, "people_vaccinated"].iloc[0]
#             else: 
#                 vaccine_people_vaccinated[sub_vaccine] = total_vaccinated
        

In [18]:
#df_vaccine_people_vaccinated.to_csv("People Vaccinated by Vaccine.csv")

# Total Vaccinations

## (we'll use people_vaccinated but will leave this here nalang also :>) 


In [19]:
df_country_total_vac = df_country_vaccinations.groupby(["country"], as_index = False)["total_vaccinations"].max()

In [20]:
df_country_total_vac

country  total_vaccinations
0    Afghanistan             54000.0
1        Albania             64075.0
2        Algeria             75000.0
3        Andorra              9288.0
4         Angola             87022.0
..           ...                 ...
149      Uruguay            509969.0
150    Venezuela             12194.0
151      Vietnam             44000.0
152        Wales           1766098.0
153     Zimbabwe             65466.0

[154 rows x 2 columns]

In [21]:
df_country_total_vac.to_csv("country_total_vac.csv")

In [22]:
df_temp_total_vac = df_country_vaccinations.groupby(["country", "vaccines"], as_index = False)["total_vaccinations"].max()

In [23]:
df_vaccine_total_vac = df_temp_total_vac.groupby(["vaccines"], as_index = False)["total_vaccinations"].sum()

In [24]:
df_vaccine_total_vac

vaccines  total_vaccinations
0                         Covaxin, Oxford/AstraZeneca          60269782.0
1                             EpiVacCorona, Sputnik V          10600000.0
2                                     Johnson&Johnson            231605.0
3           Johnson&Johnson, Moderna, Pfizer/BioNTech         140180735.0
4                                             Moderna             91918.0
5                         Moderna, Oxford/AstraZeneca             43073.0
6        Moderna, Oxford/AstraZeneca, Pfizer/BioNTech          66305639.0
7   Moderna, Oxford/AstraZeneca, Pfizer/BioNTech, ...           2589469.0
8                            Moderna, Pfizer/BioNTech          13930466.0
9                                  Oxford/AstraZeneca          10195050.0
10                Oxford/AstraZeneca, Pfizer/BioNTech          73335876.0
11  Oxford/AstraZeneca, Pfizer/BioNTech, Sinopharm...           7959682.0
12  Oxford/AstraZeneca, Pfizer/BioNTech, Sinopharm...           3006877.0
13       Oxford/AstraZeneca, Pfizer/BioNTech, Sinovac             97969.0
14     Oxford/AstraZeneca, Pfizer/BioNTech, Sputnik V           6724789.0
15              Oxford/AstraZeneca, Sinopharm/Beijing           7730998.0
16   Oxford/AstraZeneca, Sinopharm/Beijing, Sputnik V           3920460.0
17                        Oxford/AstraZeneca, Sinovac          18929465.0
18                                    Pfizer/BioNTech           4714539.0
19                 Pfizer/BioNTech, Sinopharm/Beijing            365048.0
20                           Pfizer/BioNTech, Sinovac          27389812.0
21                         Pfizer/BioNTech, Sputnik V             38853.0
22                                  Sinopharm/Beijing           1432185.0
23        Sinopharm/Beijing, Sinopharm/Wuhan, Sinovac         102417000.0
24                       Sinopharm/Beijing, Sputnik V             57183.0
25                                            Sinovac          10936110.0
26                                          Sputnik V            678927.0

# Infections WorldWide

In [25]:
df_infections_ww_ds = pd.read_csv("Infections WorldWide.csv")

In [26]:
df_infections_ww_ds

Unnamed: 0     Country,Other  Total Cases New Cases Total Deaths  \
0           NaN             World  135,754,176   464,052    2,934,228   
1           1.0               USA   31,818,468    15,696      574,989   
2           2.0            Brazil   13,375,414       NaN      348,934   
3           3.0             India   13,355,297   152,514      169,304   
4           4.0            France    4,980,501       NaN       98,395   
..          ...               ...          ...       ...          ...   
218       218.0  Marshall Islands            4       NaN          NaN   
219       219.0             Samoa            3       NaN          NaN   
220       220.0           Vanuatu            3       NaN          NaN   
221       221.0        Micronesia            1       NaN          NaN   
222         NaN            Total:  135,754,176   464,052    2,934,228   

    New Deaths Total Recovered Active Cases Serious,Critical  \
0        6,478     109,192,849   23,627,099          102,376   
1          149      24,355,083    6,888,396            9,080   
2          NaN      11,791,885    1,234,595            8,318   
3          837      12,078,240    1,107,753            8,944   
4          NaN         303,639    4,578,467            5,729   
..         ...             ...          ...              ...   
218        NaN               4            0              NaN   
219        NaN               2            1              NaN   
220        NaN               1            2              NaN   
221        NaN               1            0              NaN   
222      6,478     109,192,849   23,627,099          102,376   

    Total Cases/1M pop Deaths/ 1M pop  Total Tests Tests/1M pop     Population  
0               17,416          376.4          NaN          NaN            NaN  
1               95,695          1,729  417,052,159    1,254,291    332,500,290  
2               62,582          1,633   28,600,000      133,817    213,724,704  
3                9,605            122  255,214,803      183,547  1,390,456,911  
4               76,172          1,505   68,007,540    1,040,108     65,385,105  
..                 ...            ...          ...          ...            ...  
218                 67            NaN          NaN          NaN         59,498  
219                 15            NaN          NaN          NaN        199,431  
220                 10            NaN          470        1,503        312,643  
221                  9            NaN          NaN          NaN        115,952  
222          17,416.00          376.4          NaN          NaN            NaN  

[223 rows x 14 columns]

In [27]:
df_infections_ww_ds.loc[df_infections_ww_ds["Country,Other"] == "India", "Total Cases"]

3    13,355,297
Name: Total Cases, dtype: object

In [29]:
for country in country_list: 
    if country not in list(df_infections_ww_ds["Country,Other"]): 
        print(country)

Cote d'Ivoire
England
Guernsey
Jersey
Northern Cyprus
Northern Ireland
Saint Helena
Saint Vincent and the Grenadines
Scotland
South Korea
Turks and Caicos Islands
United Arab Emirates
United Kingdom
United States
Wales


In [30]:
data_country_infections = {"Country" : country_list}
df_country_infections = pd.DataFrame(data_country_infections)

In [31]:
name_equivalents = {"Cote d'Ivoire" : "Ivory Coast", "South Korea" : "S. Korea", "United Arab Emirates" : "UAE", "United Kingdom" : "UK",
                   "United States" : "USA", "Turks and Caicos Islands" : "Turks and Caicos"}

In [32]:
name_equivalents_old = list(name_equivalents.keys()) 
name_equivalents_new = list(name_equivalents.values())

In [33]:
df_country_infections["Country"] = df_country_infections["Country"].replace(name_equivalents_old, name_equivalents_new)

In [35]:
country_infections = {"England" : 3822032, "Guernsey" : 821, "Jersey" : 3219, "Northern Cyprus" : 49988, "Northern Ireland" : 118146, 
                     "Saint Vincent and the Grenadines" : 1791, "Scotland" : 222281, "Wales" : 210125}

In [41]:
for country in df_country_infections["Country"]: 
    if country in country_infections.keys(): 
        df_country_infections.loc[df_country_infections["Country"] == country, "Total Cases"] = country_infections[country]
    else: 
        try: 
            df_country_infections.loc[df_country_infections["Country"] == country, "Total Cases"] = df_infections_ww_ds.loc[df_infections_ww_ds["Country,Other"] == country, "Total Cases"].iloc[0]
        except: 
            pass

In [44]:
df_country_infections.dropna(subset = ["Total Cases"])

Country Total Cases
0                         Afghanistan      57,160
1                             Albania     128,155
2                             Algeria     118,251
3                             Andorra      12,497
4                              Angola      23,242
5                            Anguilla          25
6                 Antigua and Barbuda       1,182
7                           Argentina   2,497,881
8                           Australia      29,396
9                             Austria     574,755
10                         Azerbaijan     283,579
11                            Bahamas       9,364
12                            Bahrain     154,280
13                         Bangladesh     678,937
14                           Barbados       3,708
15                            Belarus     334,863
16                            Belgium     917,917
17                             Belize      12,485
18                            Bermuda       1,773
19                            Bolivia     280,649
20                             Brazil  13,375,414
21                           Bulgaria     370,179
22                           Cambodia       4,081
23                             Canada   1,050,853
24                     Cayman Islands         516
25                              Chile   1,068,522
26                              China      90,400
27                           Colombia   2,504,206
28                         Costa Rica     222,544
29                        Ivory Coast      45,089
30                            Croatia     290,899
31                             Cyprus      50,589
32                            Czechia   1,577,972
33                            Denmark     237,101
34                           Dominica         165
35                 Dominican Republic     257,186
36                            Ecuador     344,877
37                              Egypt     208,876
38                        El Salvador      65,491
39                            England     3822032
40                  Equatorial Guinea       7,059
41                            Estonia     113,736
42                     Faeroe Islands         661
43                   Falkland Islands          60
44                            Finland      81,707
45                             France   4,980,501
46                            Germany   2,983,278
47                              Ghana      91,109
48                          Gibraltar       4,277
49                             Greece     293,763
50                          Greenland          31
51                            Grenada         155
52                          Guatemala     201,295
53                           Guernsey         821
54                             Guinea      20,807
55                             Guyana      10,958
56                           Honduras     194,548
57                          Hong Kong      11,569
58                            Hungary     713,868
59                            Iceland       6,258
60                              India  13,355,297
61                          Indonesia   1,562,868
62                               Iran   2,049,078
63                            Ireland     240,643
64                        Isle of Man       1,574
65                             Israel     835,811
66                              Italy   3,754,077
67                            Jamaica      42,119
68                              Japan     499,793
69                             Jersey        3219
70                             Jordan     662,395
71                         Kazakhstan     266,677
72                              Kenya     145,184
73                             Kuwait     244,325
74                               Laos          49
75                             Latvia     107,240
76                            Lebanon     494,633
77                      Liechtenstein       2,738
78                          Lithuania     225,308
79                         Lux

## Scratch

Google Search: 

England: 3822,032
Guernsey 821
Jersey 3219
Northern Cyprus 49,988 (cyrpus)
Northern Ireland 118,146
Saint Helena
Saint Vincent and the Grenadines 1791
Scotland 222, 281
Wales 210,125

In [39]:
name_equivalents

{"Cote d'Ivoire": 'Ivory Coast',
 'South Korea': 'S. Korea',
 'United Arab Emirates': 'UAE',
 'United Kingdom': 'UK',
 'United States': 'USA',
 'Turks and Caicos Islands': 'Turks and Caicos'}

Cote d'Ivore - Ivory Coast
Northern Cyprus - Turkey? :/
Northern Ireland - Ireland
S.Korea
UAE
UK
USA